HealthcareApp/src/Healthcare.Tests/UserAuthenticationServiceTests.cs

```diff
@@ -0,0 +1,135 @@
+using Application.Abstractions;
+using Application.DTOs.Login;
+using Moq;
+using FluentAssertions;
+using Microsoft.AspNetCore.Identity;
+using Application.DTOs.Register;
+using Healthcare.Infrastructure.Persistance;
+using Infrastructure.Repository;
+using Microsoft.AspNetCore.Http;
+
+
+namespace Healthcare.Tests;
+
+public class UserAuthenticationServiceTests
+{
+    private readonly Mock<UserManager<ApplicationUser>> _userManagerMock;
+    private readonly Mock<SignInManager<ApplicationUser>> _signInManagerMock;
+    private readonly Mock<RoleManager<IdentityRole>> _roleManagerMock;
+    private readonly Mock<ITokenGeneratorService> _generatorService;
+    private readonly AuthenticationService _authService;
+
+    public UserAuthenticationServiceTests()
+    {
+        _userManagerMock = new Mock<UserManager<ApplicationUser>>(
+            Mock.Of<IUserStore<ApplicationUser>>(), null, null, null, null, null, null, null, null);
+
+        _signInManagerMock = new Mock<SignInManager<ApplicationUser>>(
+            _userManagerMock.Object, Mock.Of<IHttpContextAccessor>(),
+            Mock.Of<IUserClaimsPrincipalFactory<ApplicationUser>>(), null, null, null, null);
+
+        _roleManagerMock = new Mock<RoleManager<IdentityRole>>(
+            Mock.Of<IRoleStore<IdentityRole>>(), null, null, null, null);
```

Comment:
Very good tests, good job finding how to mock the implementations of SignIn services.
Please take a look at the alternative way of doing this, [using decorator pattern](https://github.com/IT-Sadok/nazarr43/pull/6/files#diff-0d08a72f3e188d81ebb5cadd4cd07331fb9eb69ba42c68a084fcefec190a3841)